src/charm.py

```diff
@@ -282,9 +282,7 @@ def _create_env(self):
             "GOOGLE_SECRET": self.config["google-client-secret"],
             "OAUTH_DOMAIN": self.config["oauth-domain"],
             "OAUTH_ADMIN_EMAIL": self.config["oauth-admin-email"],
-            "SELF_REGISTRATION_ROLE": self.config[
-                "self-registration-role"
-            ].value,
+            "SELF_REGISTRATION_ROLE": self.config["self-registration-role"],
```

Comment:
Any validation you think you should run on the provided value?

Reply:
Other than not being empty which is handled by the default and structure config the Superset role names can be anything (special characters allowed) so any validation I put here would be potentially restrictive on others. We either go Enum route and align it with our groups or we keep it open so it's more usable by others. I think you were right before - I don't think I should put limitations on top of Superset that do not exist in the application.

Reply:
@kelkawi-a I've realised while this value can accept anything the role does need to exist in superset. Added some validation for this and requesting a re-review.